src/state_event.rs

```diff
@@ -5,16 +5,21 @@ use crate::{
         EventReader,
     },
     derive::EventReader,
+    input::InputEvent,
     renderer::Event,
     ui::UiEvent,
 };
 
 /// The enum holding the different types of event that can be received in a `State` in the handle_event method.
+/// This assumes that you used `String` as the identifier for the `InputBundle`. If this is not the
```

Comment:
how about type parameterize it with a `String` default?

Reply:
Generics don't work with the macro :(

Reply:
wanna venture into proc macro land to enable that? It's good to learn (and it's not too scary, promise!)

Reply:
Agreed, I think that would work nicely. @jojolepro I ventured into proc macro land for the `WidgetId` derive, so if you need any pointer you can hit me up too.

Otherwise, this looks fine to me =)

Reply:
it supports it now 🚀!